hera_qm/xrfi.py

```diff
@@ -222,6 +224,223 @@ def robust_divide(num, den):
+    eigenval_cutoff: array-like, default=[1e-9]
+        List of sinc_matrix eigenvalue cutoffs to use for included DPSS modes.
+    suppression_factors: array-like, default=[1e-9]
+        Specifies the fractional residuals of model to leave in the data. For example, 1e-6 means that the filter
+        will leave in 1e-6 of data fitted by the model.
+    cache: dictionary, default={}
+        dictionary for caching fitting matrices.
+
+    Returns:
+    -------
+    wgts: np.ndarray
+        Array containing weights for each antenna with values of either zero or one with
+        zero identifying channels found to have RFI
+    """
+    if len(suppression_factors) == 1 and len(filter_centers) > 1:
+        suppression_factors = len(filter_centers) * suppression_factors
+         
+    if len(eigenval_cutoff) == 1 and len(filter_centers) > 1:
+        eigenval_cutoff = len(filter_centers) * eigenval_cutoff
+
+    if wgts is None:
+        wgts = np.ones_like(data)
+
+    else:
+        wgts = deepcopy(wgts)
+
+    # Compute first-pass model and residuals
+    model, _, _ = dspec.fourier_filter(
+        freqs, data, wgts, filter_centers, filter_half_widths, mode=mode,
+        suppression_factors=suppression_factors, eigenval_cutoff=eigenval_cutoff,
+        cache=cache
+    )
+    res = data - model
+
+    # Compute modified z-score
+    sigma = np.median(
+        np.abs(res - np.median(res, keepdims=True, axis=1)), axis=1, keepdims=True,
+    )
+    sigma /= 0.675
+    wgts = np.where(np.abs(res) < nsig * sigma, 1, 0)
+
+    # Recompute model with first pass weights
+    model, _, _ = dspec.fourier_filter(
+        freqs, data, wgts, filter_centers, filter_half_widths, mode=mode,
+        suppression_factors=suppression_factors, eigenval_cutoff=eigenval_cutoff,
+        cache=cache
+    )
+    res = data - model
+
+    # Use noise standard deviation given or estimate it from residuals
+    if noise is not None:
+        sigma = noise
+
+    else:
+        sigma = np.median(
+            np.abs(res - np.median(res, keepdims=True, axis=1)), axis=1, keepdims=True,
+        )
+        sigma /= 0.675
+
+    # Determine weights
+    wgts = np.where(np.abs(res) <= nsig * sigma, 1, 0)
+    return wgts
+
+
+def channel_diff_flagger(data, noise, nsig=10, kernel_widths=[1, 2, 4], wgts=None):
+    """
+    Flags auto-correlations using channel differencing kernels
+
+    Parameters:
+    ----------
+    data: np.ndarray
+        2D data array of the shape (time, frequency)
+    noise: np.ndarray
+        2D array for containing an estimate of the noise standard deviation. Must be the same shape as the data
+    nsig: float, default=10
+        The number of sigma in the metric above which to flag pixels.
+    kernel_width: list
+        Widths of the convolution kernels used to produce model
+    wgts: np.ndarray, default=None
+        2D array of real weights. Must be the same shape as data.
+
+    Returns:
+    -------
+    wgts: np.ndarray
```

Comment:
Same comment as before about bools

Reply:
Modified the name of this similarly to above